pandas/core/algorithms.py

```diff
@@ -101,13 +101,7 @@ def _ensure_data(values, dtype=None):
         elif is_object_dtype(values) and dtype is None:
             return ensure_object(np.asarray(values)), "object", "object"
         elif is_complex_dtype(values) or is_complex_dtype(dtype):
-
-            # ignore the fact that we are casting to float
-            # which discards complex parts
-            with catch_warnings():
-                simplefilter("ignore", np.ComplexWarning)
-                values = ensure_float64(values)
-            return values, "float64", "float64"
+            raise TypeError("Complex data types not supported...Coercing to object")
```

Comment:
better to do the object-coersion here than fallback to the except branch.  clearer